apiserver/facades/client/action/action.go

```diff
@@ -118,21 +133,44 @@ func (a *ActionAPI) Actions(ctx context.Context, arg params.Entities) (params.Ac
 	}
 
 	response := params.ActionResults{Results: make([]params.ActionResult, len(arg.Entities))}
+	// Temporary map to hold the index of each action UUID in the results slice.
+	results := make(map[coreaction.UUID]int)
+
+	// Try to parse all the tags first.
 	for i, entity := range arg.Entities {
-		currentResult := &response.Results[i]
 		tag, err := names.ParseTag(entity.Tag)
 		if err != nil {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+			response.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
 			continue
 		}
-		_, ok := tag.(names.ActionTag)
+		actionTag, ok := tag.(names.ActionTag)
 		if !ok {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+			response.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
 			continue
 		}
+		results[coreaction.UUID(actionTag.Id())] = i
+	}
 
-		currentResult.Error = apiservererrors.ServerError(errors.NotSupportedf("actions in Dqlite"))
+	// Bulk fetch the parsed action UUIDs.
+	actionUUIDs := transform.MapToSlice(results, func(k coreaction.UUID, idx int) []coreaction.UUID { return []coreaction.UUID{k} })
+	actions, err := a.operationService.GetActions(ctx, actionUUIDs)
+	if err != nil {
+		return response, apiservererrors.ServerError(err)
```

Comment:
This error becomes incorrect for the results.

Reply:
Done, see 7e13d6c661a28b73a891c5fe2679d083f15046d0.